### PR TITLE
Add NonCompliance Model

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
For Non Compliance Work

# What has changed

# How to test?
With Case Proc and ATs from ticket

# Links
https://trello.com/c/bEEgKqZY/1404-nc-import-letter-file-1-case-processor-work-8

### CODE REVIEW

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Breaks the cardinal rule of mixing DTO and entities - the DB needs to change independently of the DTOs, because case data stored in the DB would require patching if there's changes, and equally queues would need draining if DTOs got changed... all of which is horrible if the models are mixed and muddled

File: src/main/java/uk/gov/ons/census/caseapisvc/model/dto/NonComplianceType.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package uk.gov.ons.census.caseapisvc.model.dto;
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/model/dto/NonComplianceType.java]: why does this live in the DTO package and not the entity package?

**Review by NickGrantX — CHANGES_REQUESTED** [review]
pls move NCT enum to entity package

File: src/main/java/uk/gov/ons/census/caseapisvc/model/dto/NonComplianceType.java (hunk 1 above)
Comment (LukeLoze1) [reply src/main/java/uk/gov/ons/census/caseapisvc/model/dto/NonComplianceType.java]: it was lonely before

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Please move enum into entity package

File: src/main/java/uk/gov/ons/census/caseapisvc/model/dto/NonComplianceType.java (hunk 1 above)
Comment (NickGrantX) [reply src/main/java/uk/gov/ons/census/caseapisvc/model/dto/NonComplianceType.java]: please move it into the package with the entity which uses it

Comment (LukeLoze1) [reply src/main/java/uk/gov/ons/census/caseapisvc/model/dto/NonComplianceType.java]: 'git push' found and used.

**Review by NickGrantX — APPROVED** [review]

**Review by NickGrantX — CHANGES_REQUESTED** [review]
You forgot to add `SELECTED_FOR_NON_COMPLIANCE` to the event types, which will cause the Case API to fail when it retrieves an NC case

**Review by NickGrantX — APPROVED** [review]

**Review by hbrace — APPROVED** [review]

**Pull request merged**